File: workspace/2023-02-20/fibonacci-number.prolog/fibonacci-number.pl

fibonacci(0, 0) :-
    true,
    !.
fibonacci(1, 1) :-
    true,
    !.
fibonacci(N, Fn) :-
    N > 1,
    is(A, N - 2),
    is(B, N - 1),
    fibonacci(A, Fa),
    fibonacci(B, Fb),
    is(Fn, Fa + Fb).

File: workspace/2023/2023-03-04/fibonacci-number.rust/fibonacci-number.rs

fn main()
{
    for x in (0..10).map(fibonacci) {
        println!("{}", x);
    }
}

fn fibonacci(n: i32) -> i32
{
    let mut a = 0;
    let mut b = 1;

    for _ in 0..n {
        b = b + a;
        a = b - a;
    }

    a
}

